packages/libs/components/src/plots/VolcanoPlot.tsx

```diff
@@ -0,0 +1,272 @@
+import { PlotProps } from './PlotlyPlot';
+
+import { significanceColors } from '../types/plots';
+import { VolcanoPlotData } from '../types/plots/volcanoplot';
+import { NumberRange } from '../types/general';
+import {
+  XYChart,
+  Tooltip,
+  Axis,
+  Grid,
+  GlyphSeries,
+  Annotation,
+  AnnotationLineSubject,
+} from '@visx/xychart';
+import { Group } from '@visx/group';
+import { max, min } from 'lodash';
+
+export interface VolcanoPlotProps extends PlotProps<VolcanoPlotData> {
+  /**
+   * Used to set the fold change thresholds. Will
+   * set two thresholds at +/- this number.
+   */
+  log2FoldChangeThreshold: number;
+  /** Set the threshold for significance. */
+  significanceThreshold: number;
+  /** x-axis range:  */
+  independentAxisRange?: NumberRange;
+  /** y-axis range: */
+  dependentAxisRange?: NumberRange;
+  /**
+   * Array of size 2 that contains a label for the left and right side
+   * of the x axis. (Not yet implemented). Expect this to be passed by the viz based
+   * on the type of data we're using (genes vs taxa vs etc.)
+   */
+  comparisonLabels?: Array<string>;
+  /** What is this plot's name? */
+  plotTitle?: string;
+  /** marker color opacity: range from 0 to 1 */
+  markerBodyOpacity?: number;
+}
+
+const EmptyVolcanoPlotData: VolcanoPlotData = {
+  foldChange: [],
+  pValue: [],
+  adjustedPValue: [],
+  pointId: [],
+};
+
+interface DataPoint {
+  foldChange: string;
+  pValue: string;
+  adjustedPValue: string;
+  pointId: string;
+  color: string;
+}
+
+/**
+ * The Volcano Plot displays points on a (magnitude change) by (significance) xy axis.
+ * It also colors the points based on their significance and magnitude change.
+ */
+function VolcanoPlot(props: VolcanoPlotProps) {
+  const {
+    data = EmptyVolcanoPlotData,
+    independentAxisRange,
+    dependentAxisRange,
+    significanceThreshold,
+    log2FoldChangeThreshold,
+    markerBodyOpacity,
+    ...restProps
+  } = props;
+
+  /**
+   * Find mins and maxes of the data and for the plot
+   */
+
+  const dataXMin = min(data.foldChange.map(Number));
+  const dataXMax = max(data.foldChange.map(Number));
+  const dataYMin = min(data.pValue.map(Number));
+  const dataYMax = max(data.pValue.map(Number));
+
+  // Determine mins, maxes of axes in the plot.
+  // These are different than the data mins/maxes because
+  // of the log transform and the little bit of padding.
+  //
+
+  let xMin: number;
+  let xMax: number;
+  let yMin: number;
+  let yMax: number;
+
+  // Log transform for plotting, and add a little margin for axes
+  if (dataXMin && dataXMax) {
+    xMin = Math.log2(dataXMin);
+    xMax = Math.log2(dataXMax);
+    xMin = xMin - (xMax - xMin) * 0.05;
```

Comment:
**nit/suggestion**: 0.05 is the measure of statistical significance, right? Maybe we can save that to a constant called `STATISTICAL_SIGNIFICANCE_THRESHOLD`?